os/userspace/up_userspace.c

```diff
@@ -96,7 +96,7 @@ const struct userspace_s userspace __attribute__((section(".userspace"))) = {
 	.ram_start = &_sapp_ram,
 	.ram_end = &_eapp_ram,
 #ifndef __COMMON_BINARY__
-	.entry = main,
+	.entry =  (void (*)(int,  char **))main,
```

Comment:
Change the type of variable entry in "struct userspace_s" as type main_t. Then use typecast main_t here.